src/ApplePayGateway.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Omnipay\Vindicia;
+
+/**
+ * Vindicia ApplePay Gateway
+ * This Apple Pay Gateway provides the functionality to retrieve an opaque payment session object from Apple.
+ * 
+ * Apple Pay on the web:
+ * Apple Pay is available on all iOS devices with a Secure Element—an industry-standard, certified chip
+ * designed to store payment information safely. On macOS, users must have an Apple Pay-capable iPhone or Apple Watch
+ * to authorize the payment, or a MacBook Pro with Touch ID.
+ *
+ * Business Chat:
+ * Business Chat let's customers chat directly with your business using the Messages app on their device.
+ * through their CSP provider that implments a REST API to enables you to request a payment through Apple Pay.
+ *
+ * You use the same username and password as you do with the regular Vindicia
+ * gateway.
+ *
+ * Example:
+ * <code>
+ *   // set up the gateway
+ *   $gateway = \Omnipay\Omnipay::create('Vindicia_ApplePay');
+ *   $gateway->setUsername('your_username');
+ *   $gateway->setPassword('y0ur_p4ssw0rd');
+ *   $gateway->setPemCertPath('./certs/path_to_cert.crt.pem');
+ *   $gateway->setKeyCertPath('./certs/path_to_cert.key.pem');
+ *   $gateway->setKeyCertPassword('your_key_cert_password');
+ *   $gateway->setTestMode(false);
+ *
```

Comment:
That's why I wanted to show the flow with the standard ApplePayJS docs code.

With some of your suggestions, I'm worried about these docs becoming ApplePayJS tutorials, because that's not what this is for. It's just one piece of the whole API and can be implemented many different ways. I like the suggestion of listing what should happen "user clicks button, payment sheet is loaded... etc" and then where parts of the gateway fits in.

I also can't go into detail about the completeAuthorize and capture because I haven't done that yet. This is one of those issues we spoke about with the tickets, I'm going to have to come back to this one later. So, I can make most of the changes you've suggested but I'm going to have to come back to this and update it once I've gone through the entire flow.

Reply:
I agree that this should not be an ApplePayJS tutorial. If someone were following an ApplePayJS tutorial, they just need to be able to tell roughly where these different steps are supposed to fit in the code. So yeah, just adding brief notes about what is supposed to happen on the front end side is totally fine.

Sure, that's fine to add the docs for completeAuthorize and capture later